background.ts

```diff
@@ -749,17 +764,121 @@ function newK8sManager() {
 
 /**
  * Implement the methods the HttpCommandServer needs to service its requests.
- * These methods should be thin wrappers around existing functionality in the rest of the backend.
- * Getters normally return strings, either JSON strings or scalars.
- * Setters normally return either `true | false`, or possibly `true | error`.
- * The `requestShutdown` is a special case that never returns.
+ * These methods do two things:
+ * 1. Verify the semantics of the parameters (the server just checks syntax).
+ * 2. Provide a thin wrapper over existing functionality in this module.
+ * Getters, on success, return status 200 and a string that may be JSON or simple.
+ * Setters, on success, return status 202, possibly with a human-readable status note.
+ * The `requestShutdown` method is a special case that never returns.
  */
 class BackgroundCommandWorker implements CommandWorkerInterface {
   getSettings() {
     return JSON.stringify(cfg, undefined, 2);
   }
 
-  requestShutdown() {
+  /**
+   * On failure, returns status 400, with a human-readable error message.
+   * @param newSettings
+   */
+  async updateSettings(newSettings: UpdatableSettings): Promise<[string, string]> {
+    const newConfig: RecursivePartial<settings.Settings> = {};
+    let desiredVersion: string;
+    let versions: string[] = [];
+    let m: any;
+    let needToUpdate = false;
+
+    for (const k in newSettings) {
+      let v: string|boolean = newSettings[k];
+      let desiredEngine: settings.ContainerEngine;
+
+      switch (k) {
+      case 'kubernetes-version':
```

Comment:
Tempting to convert it into a dispatch table of functions, because there's so much repetition.  I've made a note for that -- will submit a separate issue-less PR if I do that.